messages/teleporter/message_handler.go

```diff
@@ -183,32 +183,34 @@ func (m *messageHandler) ShouldSendMessage(destinationClient vms.DestinationClie
 		return false, nil
 	}
 
-	if decision, err := m.getDeciderDecision(); err != nil {
+	if decision, err := m.getShouldSendMessageFromDecider(); err != nil {
```

Comment:
I'm still not a huge fan of this control flow. The `return true, nil` at the end of this function is only reachable if `m.getShouldSendMessageFromDecider` errors, so IMO the code would be more readable if we returned at that point:

```
decision, err := m.getShouldSendMessageFromDecider()
if err != nil {
    // If the decider errors, default to true
    m.logger.Warn(
        "Error delegating to decider",
        zap.String("warpMessageID", m.unsignedMessage.ID().String()),
        zap.String("teleporterMessageID", teleporterMessageID.String()),
    )
    return true, nil
}

if !decision {
    m.logger.Info(
        "Decider rejected message",
        zap.String("warpMessageID", m.unsignedMessage.ID().String()),
        zap.String("teleporterMessageID", teleporterMessageID.String()),
        zap.String("destinationBlockchainID", destinationBlockchainID.String()),
    )
}

return decision, nil
```

This also removes the nested `if`. Lmk if you have other thoughts here.

Reply:
Yeah, this is the flow I prefer when a function returns an `err` and a `bool`

Reply:
thanks for pushing to make this great. addressed in https://github.com/ava-labs/awm-relayer/pull/344/commits/8ed81129aaeb1329a0eb2ef86e84542806ab392a